includes/class-wcs-att-cart.php

```diff
@@ -242,9 +250,22 @@ public static function load_cart_item_data_from_session( $cart_item, $item_sessi
 	public static function apply_subscription_scheme( $cart_item ) {
 
 		if ( self::is_supported_product_type( $cart_item ) ) {
-			$scheme_key = self::get_subscription_scheme( $cart_item );
-			if ( null !== $scheme_key ) {
-				WCS_ATT_Product_Schemes::set_subscription_scheme( $cart_item[ 'data' ], $scheme_key );
+
+			$scheme_to_apply = self::get_subscription_scheme( $cart_item );
+
+			if ( null !== $scheme_to_apply ) {
+
+				// Attempt to apply scheme.
+				WCS_ATT_Product_Schemes::set_subscription_scheme( $cart_item[ 'data' ], $scheme_to_apply );
+
+				// Grab the applied scheme.
+				$applied_scheme = WCS_ATT_Product_Schemes::get_subscription_scheme( $cart_item[ 'data' ] );
+
+				// If the scheme was not applied sucessfully, then it was probably deleted, or something fishy happened.
+				if ( $scheme_to_apply !== $applied_scheme ) {
+					// In this case, simply ensure that no scheme is set on the object and handle the mismatch later.
+					WCS_ATT_Product_Schemes::set_subscription_scheme( $cart_item[ 'data' ], null );
+				}
```

Comment:
@thenbrent since we are validating on `WCS_ATT_Product_Schemes ::set_subscription_scheme` as well as on `WCS_ATT_Product_Schemes ::get_subscription_scheme`, I agree this is probably not needed! 👍